src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -2692,4 +2694,323 @@ public void terminate() throws Exception {
+                        stmt.executeUpdate("INSERT INTO " + tableName + " (NAME) VALUES ('test')");
+                    }
+
+                }
+                con.commit();
+            }
+        }
+
+        /**
+         * Tests executeUpdate for Insert followed by getGenerateKeys
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testExecuteUpdateInsertAndGenKeys() throws Exception {
+            try (Connection con = getConnection()) {
+                try(Statement stmt = con.createStatement()) {
+                    String sql = "INSERT INTO " + tableName + " (NAME) VALUES('test')";
+                    List<String> lst = Arrays.asList("ID");
+                    String[] arr = lst.toArray(new String[0]);
+                    stmt.executeUpdate(sql, arr);
+                    try (ResultSet generatedKeys = stmt.getGeneratedKeys()) {
+                        if (generatedKeys.next()) {
+                            int id = generatedKeys.getInt(1);
+                            assertEquals(id, 4, "id should have been 4, but received : " + id);
+                        }
+                    }
+                }
+            } catch (SQLException e) {
+                fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
+            }
+        }
+
+        /**
+         * Tests execute for Insert followed by getGenerateKeys
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testExecuteInsertAndGenKeys() throws Exception {
+            try (Connection con = getConnection()) {
+                try(Statement stmt = con.createStatement()) {
+                    String sql = "INSERT INTO " + tableName + " (NAME) VALUES('test')";
+                    List<String> lst = Arrays.asList("ID");
+                    String[] arr = lst.toArray(new String[0]);
+                    stmt.execute(sql, arr);
+                    try (ResultSet generatedKeys = stmt.getGeneratedKeys()) {
+                        if (generatedKeys.next()) {
+                            int id = generatedKeys.getInt(1);
+                            assertEquals(id, 4, "generated key should have been 4");
+                        }
+                    }
+                }
+            } catch (SQLException e) {
+                fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
+            }
+        }
+
+        /**
+         * Tests execute for Insert followed by select
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testExecuteInsertAndSelect() throws Exception {
+
+            try (Connection con = getConnection()) {
+                try(Statement stmt = con.createStatement()) {
+                    boolean retval = stmt.execute("INSERT INTO " + tableName +" (NAME) VALUES('test') SELECT NAME FROM " + tableName + " WHERE ID = 1");
+                    do {
+                        if (retval == false) {
```

Comment:
Also replace in all other methods where this is used.
```suggestion
                        if (!retval) {
```

Reply:
done